compiler/hash-ast/src/ast.rs

```diff
@@ -579,8 +594,8 @@ pub enum ExpressionKind<'c> {
     Variable(VariableExpr<'c>),
     /// A property access.
     PropertyAccess(PropertyAccessExpr<'c>),
-    /// A reference expression.
-    Ref(AstNode<'c, Expression<'c>>),
+    /// A reference expression with a flag denoting whether it is a raw ref or not
+    Ref(AstNode<'c, Expression<'c>>, bool),
```

Comment:
Yes this could be changed, but i think there might be alot of code duplication in some parts when dealing with refs. Admittedly, it might be better to distinguish the two when it comes to later stages where the difference does actually matter.

Reply:
Then we might want a `RefType` enum at least, because we will need more than raw/non-raw references (e.g. weak references, mutable references in the future).